File: app/lib/models/evenement_request.dart
class EvenementRequest {
  final String titre;
  final String? description;
  final DateTime? date;
  final String? type;
  /// Lieu de rencontre / localisation (adresse ou nom du lieu).
  final String? lieu;
  final double? latitude;
  final double? longitude;
  final int groupeId;

  EvenementRequest({
    required this.titre,
    this.description,
    this.date,
    this.type,
    this.lieu,
    this.latitude,
    this.longitude,
    required this.groupeId,
  });

  Map<String, dynamic> toJson() {
    return {
      'titre': titre,
      if (description != null && description!.isNotEmpty) 'description': description,
      if (date != null) 'date': date!.toUtc().toIso8601String(),
      if (type != null && type!.isNotEmpty) 'type': type,
      if (lieu != null && lieu!.isNotEmpty) 'lieu': lieu,
      if (latitude != null) 'latitude': latitude,
      if (longitude != null) 'longitude': longitude,
      'groupeId': groupeId,
    };
  }
}

File: demo/src/main/java/com/example/demo/dto/EvenementRequest.java
package com.example.demo.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class EvenementRequest {

    @NotBlank(message = "Le titre est requis")
    private String titre;

    private String description;

    private Instant date;

    private String type;

    /** Lieu de rencontre / localisation (adresse ou nom du lieu). */
    private String lieu;

    private Double latitude;
    private Double longitude;

    @NotNull(message = "L'identifiant du groupe est requis")
    private Long groupeId;
}
